### ISSUE TITLE
Problème de connexion

### ISSUE THREAD

**Issue by abraracourcix67** [MateoGreil/homeassistant-comwatt#7]
Bonjour

Je pense être sur une version antérieure à la Gen4 du coup mes codes d'accès ne fonctionnent pas.Je me connecte avec mail et mdp.
![Capture d'écran 2023-12-16 123054](https://github.com/MateoGreil/homeassistant-comwatt/assets/153768258/75b5e069-4f1e-4fa0-b78b-5f9005db5fb9)
Je vous mets l'image de l'appli que j'utilise.
![Capture d'écran 2023-12-16 123636](https://github.com/MateoGreil/homeassistant-comwatt/assets/153768258/42ad09c2-b43a-4b2d-8dbd-f1edec6f2010)

Cordialement
Franck

**Comment by ZoomeoTooknor** [comment]
Hello Mateo,

First, thanks for you job.
I Have the same issue than Franck, with the message "Failed to connect".
Can you help me please ?

Best regards

**Comment by MateoGreil** [comment]
Bonjour,
Pouvez vous m'envoyer les logs lié à cette erreur s'il vous plaît ?

Vous les trouverez dans :
Paramètres -> Système -> Journaux, dans la partie core, vous pouvez chercher comwatt ça devrait vous afficher le log en question. S'il ne s'affiche pas ne pas hésiter a reproduire le bug puis regarder à nouveau les logs.

**Comment by abraracourcix67** [comment]
Bonsoir

Voilà ce que j'ai trouvé:
Logger: homeassistant.loader
Source: loader.py:581
First occurred: 22:14:10 (5 occurrences)
Last logged: 22:14:10

We found a custom integration tapo which has not been tested by Home Assistant. This component might cause stability problems, be sure to disable it if you experience issues with Home Assistant
We found a custom integration tapo_control which has not been tested by Home Assistant. This component might cause stability problems, be sure to disable it if you experience issues with Home Assistant
We found a custom integration hacs which has not been tested by Home Assistant. This component might cause stability problems, be sure to disable it if you experience issues with Home Assistant
We found a custom integration comwatt which has not been tested by Home Assistant. This component might cause stability problems, be sure to disable it if you experience issues with Home Assistant
We found a custom integration google_home which has not been tested by Home Assistant. This component might cause stability problems, be sure to disable it if you experience issues with Home Assistant

Merci pour votre aide.

**Comment by abraracourcix67** [comment]
Je viens de relancer.
2023-12-17 22:51:05.626 WARNING (SyncWorker_2) [homeassistant.loader] We found a custom integration comwatt which has not been tested by Home Assistant. This component might cause stability problems, be sure to disable it if you experience issues with Home Assistant

**Comment by ZoomeoTooknor** [comment]
Bonjour Mateo,

Voici mon log :

2023-12-17 07:46:07.606 WARNING (SyncWorker_3) [homeassistant.loader] We found a custom integration comwatt which has not been tested by Home Assistant. This component might cause stability problems, be sure to disable it if you experience issues with Home Assistant

**Comment by MateoGreil** [comment]
Ah ok, j'ai probablement pas mis de log en fait 😅

Je regarde pour en ajouter ou trouver le problème des que j'ai du temps.

**Comment by MateoGreil** [comment]
Effectivement, c'est parce que j'utilise l'api `https://energy.comwatt.com/api`
Tu utilises sur ta capture d'ecran `go.comwatt.com`, il semble que cette application n'utilise pas la meme API.

N'est-ce pas possible pour les versions anterieurs a la gen4 d'utiliser energy.comwatt.com ?

**Comment by abraracourcix67** [comment]
Bonjour

Impossible de me connecter sur [energy](https://energy.comwatt.com)

![image](https://github.com/MateoGreil/homeassistant-comwatt/assets/153768258/e0c9564d-24fc-49d2-90c0-c543ff4074e2)

Tu penses pouvoir faire quelque chose?

Merci quand même pour le temps passé.

Franck

**Comment by MateoGreil** [comment]
Yes, j'ai ajouté la possibilité d'utiliser l'api go.comwatt.com/api [ici](https://github.com/MateoGreil/python-comwatt-client/pull/2),
Il faudrait maintenant l'essayer,
Le mieux serait que je vous fasse les modifications, mais que vous les appliquiez vous-meme avec un editeur de texte, afin d'etre certain que ca fonctionne avant que je release cette nouvelle version.
Si je vous fait les modifications sur l'integration, saurez-vous modifier vos fichiers avec l'add-on `File editor` par exemple ?

**Comment by ZoomeoTooknor** [comment]
Pour ma part, je devrais pouvoir le faire.
Il faut juste quelques explications et nous indiquer ce que l'on doit modifier

**Comment by MateoGreil** [comment]
Je prepare ca quand j'ai du temps

**Comment by abraracourcix67** [comment]
Bonjour
Merci pour votre réactivité.
Je suis débutant et vraiment pas sur de savoir faire les modifs, je pense que cela vaut mieux.

Encore merci

**Comment by MateoGreil** [comment]
Hello, je viens de regarder et j'ai bien l'impression que ce n'est pas la même API. Donc changer l'URL ne suffira pas, il va falloir refaire tout le travail que j'avais fait pour comprendre quel endpoint permet d'avoir quelle info.

J'ai tout de même codé les modifications. Si @ZoomeoTooknor tu peux essayer et me confirmer que ça ne fonctionne pas : https://github.com/MateoGreil/homeassistant-comwatt/pull/8/files (il faut appliquer les mêmes changements, redémarrer Home Assistant et retenter la création).

Ce qu'il risque de se passer à mon avis : l'authentification va réussir, mais tes appareils ne vont pas être ajoutés. J'attends ton retour pour avancer sur le sujet.

**Comment by ZoomeoTooknor** [comment]
@MateoGreil , peux-tu m'indiquer comment faire les modifications ? Je ne vois pas les sources dans Home Assistant.
D'autre part, j'ai regardé tes modifications de code, et je ne vois pas où tu as changé l'URL de l'API.

**Comment by ZoomeoTooknor** [comment]
Hello @MateoGreil,

J'ai bien apporté les modifications, mais effectivement, malgré le fait de cocher "go" sur l'écran de configuration, le module a bien été activé, mais on ne récupère la liste des appareils :
<img width="376" alt="screenshot" src="https://github.com/MateoGreil/homeassistant-comwatt/assets/153979027/8ca51769-7ea9-4c18-93a6-612be7a41d30">
<img width="345" alt="screenshot2" src="https://github.com/MateoGreil/homeassistant-comwatt/assets/153979027/a64ea312-311b-428c-89b3-ca3a8d7c1011">

**Comment by ZoomeoTooknor** [comment]
@MateoGreil, si je comprends bien, le souci vient du fichier sensor.py où on ne récupère pas les sites ou les devices...
Je peux peut-être te donner un coup de main, je suis développeur logiciel de métier, mais jamais fais python.
Comment est-ce que je peux debugger du python ? 
Quel éditeur de code utilises tu ?

**Comment by MateoGreil** [comment]
En fait le problème vient du fait que l'API go et energy sont différentes.
Si tu veux avancer sur le sujet, il va falloir regarder sur une page internet tous les échanges entre le navigateur et le serveur, pour essayer de comprendre comment fonctionne cette API.
Vu que je n'ai pas de box gen3, je ne vais rien pouvoir faire.

**Comment by ZoomeoTooknor** [comment]
@MateoGreil, et si je te communiques en privé mes identifiants ?

**Comment by MateoGreil** [comment]
C'est pas très sain de partager ses identifiants avec un inconnu,
Mais oui, on peut faire comme ça.
Après honnêtement sachez que je vais prioriser les autres problèmes a celui ci.

**Comment by ZoomeoTooknor** [comment]
@MateoGreil, je comprends et tu as raison, mais les données perso y sont très limitées.
Sinon, peux-tu juste me dire comment analyser le trafic entre le navigateur et le serveur ?
via la console navigateur de chrome ?
par wireshark ?

**Comment by ZoomeoTooknor** [comment]
@abraracourcix67 , pour info. sur ta première capture d'écran, tu as pensé à cacher ton nom en haut à droite, mais pas dans le fil d'ariane à gauche de la fenêtre

**Comment by ZoomeoTooknor** [comment]
@MateoGreil, j'ai extrait les trames et les URL appelées à la connexion.
Je ne sais pas comment ça se passe avec une GEN4, mais là voici le déroulé

Authentification : https://go.comwatt.com/api/v1/authent
Récupération infos utilisateurs : https://go.comwatt.com/api/users/authenticated
Récupération identifiant box GEN3 : https://go.comwatt.com/api/indepboxes?ownerid=
Récupération infos box : https://go.comwatt.com/api/indepboxes/byMacAddress/{MacAdressBox}

Après j'ai un ensemble d'appels aux URL suivants :
wss://frontage.go.comwatt.com/ws
https://go.comwatt.com/api/products/1
https://go.comwatt.com/api/devices?indepbox_id=



Tu crois qu'avec ça tu peux faire quelque chose ?

**Comment by ZoomeoTooknor** [comment]
En utilisant Postman je suis parvenu à exploiter quelque chose.
Concernant l'intégration dans ton code, j'ai un peu plus de mal.
[reponse_authenticated.json](https://github.com/MateoGreil/homeassistant-comwatt/files/13764445/reponse_authenticated.json)
[reponse_indepbox.json](https://github.com/MateoGreil/homeassistant-comwatt/files/13764447/reponse_indepbox.json)
[reponse_devices.json](https://github.com/MateoGreil/homeassistant-comwatt/files/13764448/reponse_devices.json)

**Comment by ZoomeoTooknor** [comment]
Bonjour @MateoGreil,

j'ai commencé à modifier le projet 'python-comwatt-client-main' en rajoutant les éléments suivants :
client.py :

    def get_user_details(self, owner_id):
        """
        Retrieves information about the user's details.

        Args:
            None

        Returns:
            dict: Information about the authenticated user.

        Raises:
            Exception: If an error occurs while retrieving the information.

        """

        url = f'{self.base_url}/api/indepboxes?ownerid={owner_id}'

        response = self.session.get(url)
        if response.status_code == 200:
            return response.json()
        else:
            raise Exception(f'Error retrieving user details : {response.status_code}')


    def get_go_devices(self, indepbox_id):
        """
        Retrieves a list of devices for the specified site.

        Args:
            indepbox_id (str): The ID of the box.

        Returns:
            list: A list of devices.

        Raises:
            Exception: If an error occurs while retrieving the devices.

        """

        url = f'{self.base_url}/api/devices?indepbox_id={indepbox_id}'

        response = self.session.get(url)
        if response.status_code == 200:
            return response.json()
        else:
            raise Exception(f'Error retrieving sites: {response.status_code}')

**Comment by ZoomeoTooknor** [comment]
Hello @MateoGreil,

Pour le reste des modifications, je pense qu'il faut modifier le projet homeassistant-comwatt, notamment le sensor.py ("async_setup_entry").
Est-ce que c'est bien ça ?

Au lieu de récupérer les sites, je dois appeler get_user_details, puis get_go_devices, c'est ça ?

**Comment by MateoGreil** [comment]
Hello @ZoomeoTooknor,

Le reste des endpoints utilisé dans le client python de l'API ne semble pas identique avec ceux que tu as cité [ici](https://github.com/MateoGreil/homeassistant-comwatt/issues/7#issuecomment-1868785719).
Donc je pense le plus simple est probablement de créé un nouveau client python pour l'api go, de reprendre tous les endpoints que tu as cité, et ensuite nous pourrons adapté l'integration comwatt avec ce nouveau client.

**Comment by ZoomeoTooknor** [comment]
Hello @MateoGreil

Quand tu parles du client python à reprendre, tu parles bien du projet que j'ai commencé à modifier ?
Si oui, ok, je vais repartir de celui-ci, enlever les procédures non utiles pour "Go-Comwatt", et implémenter les nouvelles.
Tout du moins je vais essayer, c'est la première fois que je fais du python ;)

Déjà, est-ce que les deux procédures que j'ai copié plus haut te semblent correctes ?

**Comment by abraracourcix67** [comment]
Bonjour
Merci pour le boulot que vous faites . Je suis désolé de ne pas pouvoir vous aider mais je n'ai aucune connaissance en programmation. Je ne pourrais que vous remonter mes observations lors des tests.

Merci encore

**Comment by ZoomeoTooknor** [comment]
Hello @MateoGreil,

Je pense avoir intégré tous les endpoints et fait le ménage dans les endpoints non utiles pour utiliser go.comwatt.
Peux-tu y jeter un oeil, et éventuellement me guider pour la suite ?
[client.py.zip](https://github.com/MateoGreil/homeassistant-comwatt/files/13852319/client.py.zip)

Merci d'avance

**Comment by MateoGreil** [comment]
Bonjour @ZoomeoTooknor ,

Etant donne que je n'ai pas de box gen3, je ne vais pas pouvoir tester ton client.
Mais si tu l'as testé et que cela te semble fonctionnel :
L'étape suivante est de créé un nouveau repo git, et un package pypi, pour ensuite pouvoir etre capable de l'utiliser dans cette integration HomeAssistant.

Si tu ne sais pas comment faire dis le moi je peux reprendre la main a partir de maintenant si tu preferes, tu as deja bien avancé sur la partie que je ne pouvais pas faire !

**Comment by ZoomeoTooknor** [comment]
Bonjour @MateoGreil,

Alors je n'ai pas fait de tests fonctionnels, car je ne vois pas comment appeler chacune de mes définitions...
Après, j'ai repris les URL que j'ai testé sous Postman, et recopier les paramètres.

Je veux bien créer un dépôt git, mais encore une fois, c'est une première pour moi.

Et je pense que ça serai dommage de créer un nouveau dépôt/projet, alors que tu as déjà un projet et un dépôt pour Comwatt.

Mais si tu penses que c'est mieux ou que tu n'as pas le temps de le faire, je peux tenter le coup.... avec un peu d'assistance de ta part ;)

**Comment by MateoGreil** [comment]
J'ai créé ce repo pour ce nouveau client :
https://github.com/MateoGreil/python-comwatt-client-legacy
Le python package :
https://pypi.org/project/comwatt-client-legacy/

Reste a faire :
Implementer son utilisation dans cette integration comwatt.

**Comment by ZoomeoTooknor** [comment]
Bonjour @MateoGreil,

Ai-je quelque chose à faire ?

**Comment by ZoomeoTooknor** [comment]
bonjour @MateoGreil ,

as-tu peux avancer sur quelque chose ?

**Comment by MateoGreil** [comment]
Hello, @ZoomeoTooknor,

J'ai a peine commencé a l'implementer : https://github.com/MateoGreil/homeassistant-comwatt/pull/12
Ce n'est absolument pas pret désolé !

**Comment by ZoomeoTooknor** [comment]
Bonjour @MateoGreil ,

est ce que tu as pu voir pour intégrer les anciennes box ?
Mon appli Comwatt ne fonctionne plus sur mon iPad. 

Merci

**Comment by sim29s** [comment]
Bonjour @MateoGreil , 

Je suis également dans le même souhait d'intégration. Je suis en BOX GEN3 via Go.comwatt.com. Si je peux aider à tester ou aider comme je peux ça serait avec plaisir. 

Simon

**Comment by ZoomeoTooknor** [comment]
Bonjour @MateoGreil ,

J'ai avancé de mon côté, mais je ne peux pas mettre à jour l'autre dépôt Git.
J'ai modifié l'appel des endpoints, et j'ai créé un fichier de tests unitaires et tout est ok.

[python-comwatt-client-legacy.zip](https://github.com/user-attachments/files/18549095/python-comwatt-client-legacy.zip)

**Comment by MateoGreil** [comment]
Hello @ZoomeoTooknor, tu peux me faire des pull requests si tu veux
En revanche je ne suis pas sur de bien comprendre ce que tu as fait :
Tu as voulu mettre l'integration HomeAssistant directement dans le python-comwatt-client-legacy ?
Ce qu'il faudrait plutot, c'est ajouter ce client sur ce repo-ci, ou bien faire une nouvelle integration HomeAssistant, c'est au choix.

**Comment by ZoomeoTooknor** [comment]
Hello @MateoGreil en fait, j'ai surtout voulut t'avancer un maximum ;).
Je ne sais pas ce qui est le mieux, je n'ai encore jamais développé de module pour home assistant

**Comment by MateoGreil** [comment]
Pour info, je viens d'ajouter les quelques diffs que tu avais sur le ComwattClient : https://github.com/MateoGreil/python-comwatt-client-legacy/commit/d84e2c03dcd22479cf2631725fa97a21351190e4

**Comment by ZoomeoTooknor** [comment]
merci @MateoGreil

**Comment by MateoGreil** [comment]
Je ne sais pas comment j'ai fait, mais je n'avais pas recu la notif de ce dernier message...

C'est top merci ! Je vais l'ajouter en lien sur ce repo !
